library/ecp.c

```diff
@@ -2626,7 +2627,7 @@ static int ecp_mul_mxz(mbedtls_ecp_group *grp, mbedtls_ecp_point *R,
     MOD_ADD(RP.X);
 
     /* Randomize coordinates of the starting point */
-    int have_rng = 1;
```

Comment:
Just move this whole statement to the top of the function, so that `have_rng` is always initialised.

That also matches what is done elsewhere in this file